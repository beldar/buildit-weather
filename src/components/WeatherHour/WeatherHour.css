.weather-hour-item {
  border-bottom: 1px solid #cbcbcb;
  padding: 1em 0;
  text-align: left;
  display: block;
}

.weather-hour-item:last-of-type {
  border: 0;
}

.weather-hour__hour {
  font-size: 1.2em;
  font-weight: bold;
}

.weather-hour__prop {
  text-decoration: underline;
  margin-right: .3em;
}
