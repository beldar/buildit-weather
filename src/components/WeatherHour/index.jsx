import React from 'react';
import './WeatherHour.css';

export default function WeatherHour(props) {
  if (!props.hour) return null;
  const hour = props.hour;

  return (
    <li className="weather-hour-item">
      <ul className="weather-hour">
        <li className="weather-hour__hour">⏱ { hour.dt_txt.split(' ')[1] }</li>
        <li><span className="weather-hour__prop">Temperature:</span> <strong>{hour.main.temp}ºC</strong></li>
        <li><span className="weather-hour__prop">Humidity:</span> <strong>{hour.main.humidity}%</strong></li>
        <li><span className="weather-hour__prop">Weather: </span> <strong>{hour.weather[0].description}</strong></li>
      </ul>
    </li>
  );
};
