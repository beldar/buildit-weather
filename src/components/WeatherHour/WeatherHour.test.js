import React from 'react';
import WeatherHour from './index';
import renderer from 'react-test-renderer';

describe('WeatherHour', () => {
  it('should not render if hour prop is not passed', () => {
    const tree = renderer.create(
      <WeatherHour hour={null} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly if hour prop is passed', () => {
    const hour = {
      dt_txt: "2017-07-06 12:00:00",
      main: {
        temp: 22.5,
        humidity: 60
      },
      weather: [{
        description: 'cloudy'
      }]
    };

    const tree = renderer.create(
      <WeatherHour hour={hour} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
