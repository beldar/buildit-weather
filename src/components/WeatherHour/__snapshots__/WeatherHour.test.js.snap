// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherHour should not render if hour prop is not passed 1`] = `null`;

exports[`WeatherHour should render correctly if hour prop is passed 1`] = `
<li
  className="weather-hour-item"
>
  <ul
    className="weather-hour"
  >
    <li
      className="weather-hour__hour"
    >
      ⏱ 
      12:00:00
    </li>
    <li>
      <span
        className="weather-hour__prop"
      >
        Temperature:
      </span>
       
      <strong>
        22.5
        ºC
      </strong>
    </li>
    <li>
      <span
        className="weather-hour__prop"
      >
        Humidity:
      </span>
       
      <strong>
        60
        %
      </strong>
    </li>
    <li>
      <span
        className="weather-hour__prop"
      >
        Weather: 
      </span>
       
      <strong>
        cloudy
      </strong>
    </li>
  </ul>
</li>
`;
